### PR TITLE
Disable docker cpuset allocation when CPUS is 0

### PR DESCRIPTION
**Pull request by meatballhat**
as advertised in the help text, but not currently true

## What is the problem that this PR is trying to fix?

The help text claims that setting `CPUS` to 0 will disable cpuset allocation, but this is not currently true.

## What approach did you choose and why?

Do like the help text claims.

## How can you test this?

- [x] staging deployment with `export TRAVIS_WORKER_DOCKER_CPUS=0`

### CODE REVIEW

File: backend/docker.go [hunk 1]
```diff
@@ -405,18 +405,23 @@ func (p *dockerProvider) Start(ctx gocontext.Context, startAttributes *StartAttr
...
-			"err":            err,
-			"cpu_set_length": len(p.cpuSets),
-			"run_cpus":       p.runCPUs,
-		}).Error("couldn't checkout CPUSets")
-		return nil, err
-	}
+	useCPUSets := p.runCPUs != uint(0)
+	cpuSets := ""
+
+	if useCPUSets {
```
Comment (meatballhat) [inline backend/docker.go]: @soulshake This guard condition ensures that the CPU set checkout never happens when using CPU shares, which also means that the CPU set size isn't checked.  Does that help explain behavior?

**Comment by meatballhat** [comment]
@soulshake There are tests that cover the defaulted CPU set size of 2, fwiw 👍

File: backend/docker.go (hunk 1 above)
Comment (soulshake) [reply backend/docker.go]: :+1:

**Comment by soulshake** [comment]
@meatballhat Sorry, I meant a test for when the CPU size is 0 :)

**Comment by meatballhat** [comment]
@soulshake do you mean like this one? https://github.com/travis-ci/worker/blob/edb4401e7ac15c73fce575e46cf196a26f2f8731/backend/docker_test.go#L321-L330

**Comment by soulshake** [comment]
@meatballhat Ah yes, that seems like it'll jolly well do :+1:

**Pull request merged**